src/Chapter3.hs

```diff
@@ -371,9 +377,28 @@ after the fight. The battle has the following possible outcomes:
  ⊛ Monster defeats the knight. In that case return -1
  ⊛ Neither the knight nor the monster wins. On such an occasion, the knight
    doesn't earn any money and keeps what they had before.
-
 -}
 
+data Entity = MkEntity {
+  health :: Int
+  , attack :: Int
+  , gold :: Int
+}
+
+type Monster = Entity
+type Knight = Entity
+
+fight :: Monster -> Knight -> Int
```

Comment:
Nice usage of type aliases!
The only issue with type aliases and reusages of the same data type for Monster and Knight in here is that you can easily mix them when calling the function:

```haskell
> fight knight monster
> fight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two type for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Entity
newtype Knight = Knight Entity
```

Reply:
Thanks so much for the explanation - having the compiler type-check the types, rather than just aliasing, sounds like a sensible idea :+1: